src/libsyntax/tokenstream.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright 2012-2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use ast::{AttrStyle};
+use codemap::{Span};
+use ext::base;
+use ext::tt::macro_parser;
+use parse::lexer::comments::{doc_comment_style, strip_doc_comment_decoration};
+use parse::lexer;
+use parse::token;
+use std::rc::Rc;
+
+//--------------------------------------------------------------------------------------//
+// TOKEN TREES                                                                          //
+//--------------------------------------------------------------------------------------//
+// DESCRIPTION                                                                          //
+// TokenTrees are syntactic forms for dealing with tokens. The description below is     //
+// more complete; in short a TokenTree is a single token, a delimited sequence of token //
+// trees, or a sequence with repetition for list splicing as part of macro expansion.   //
+//--------------------------------------------------------------------------------------//
```

Comment:
Prefer simple comment formatting to fancy - i.e. just `//` on the rhs. In fact these should probably be doc comments (`//!`)